### ISSUE TITLE
Allow using identities for the `Address` arguments

### ISSUE THREAD

**Issue by dmkozh** [stellar/soroban-tools#432]
It would be convenient to just directly use an identity whenever `Address` is needed. It could automatically sign the respective payload after preflight (if any), so that there is no need to do any additional signing.

**Comment by tyvdh** [comment]
Related https://discord.com/channels/897514728459468821/1077327403606609921

**Comment by willemneal** [comment]
@dmkozh Could you provide more details.  Do you mean `Address`'s passed in as an argument?

**Comment by dmkozh** [comment]
I mean passing identity for an `Address` argument.

For example, given a function `fn foo(addr: Address) { addr.require_auth(); }` I want to be able to invoke this with `-- --addr identity:my_identity` that will preflight `foo` and sign the required payload automatically (exact syntax up to debate).

**Comment by willemneal** [comment]
Yeah that's definitely doable.  Should this just be when it's the first argument? And could be a vector of Addresses?

**Comment by dmkozh** [comment]
No, there is nothing special about the `Address` argument (i.e. they may appear anywhere, including being struct elements etc.). So the identity syntax should apply anywhere an `Address` appears.
